### DISCUSSION TITLE
Support for non holonomic mobile manipulators

### DISCUSSION THREAD

**Discussion by benoit-robotics** [NVlabs/curobo#425] [Software Q&A]
Hello,

We have a robot with a non holonomic base (differential base). The example provided of a mobile manipulator (_franka_mobile_) has an omnidirectional base, so is not suitable for our case. The base of the robot can move along the x axis of the _base_link_ frame and rotate around the z axis.

We modified the urdf to add these extra joints to the base :

```
<link name="base_link_x"/>
<link name="rotate_link_z"/>

<joint name="rotate_z" type="revolute">
    <origin rpy="0 0 0.0" xyz="0 0 0.0"/>
    <parent link="world"/>
    <child link="rotate_link_z"/>
    <axis xyz="0 0 1"/>
    <dynamics damping="10.0"/>
    <limit effort="87" lower="-6.0" upper="6.0" velocity="1.50"/>
  </joint>

<joint name="base_x" type="prismatic">
    <origin rpy="0 0 0.0" xyz="0 0 0.0"/>
    <parent link="rotate_link_z"/>
    <child link="base_link_x"/>
    <axis xyz="1 0 0"/>
    <dynamics damping="10.0"/>
    <limit effort="87" lower="-500.0" upper="500.0" velocity="1.0"/>
  </joint>

<joint name="base_fixed" type="fixed">
    <origin rpy="0 0 0.0" xyz="0 0 0.1"/>
    <parent link="base_link_x"/>
    <child link="base_link"/>
    <axis xyz="0 0 0"/>
  </joint>
```

But the trajectory generated by Curobo in simulation are not feasible by a real robot with a differential base. For a real robot, controlling forward movement and rotation simultaneously using joint positions is not possible.
Does Curobo support motion generation for non holonomic robots ?

**Comment by balakumar-s** [comment]
Currently, we only support joints that can be implemented as a kinematic chain. Is there a way to map non-holonomic base to a kinematic chain?
